Firestore/core/src/core/query.h

```diff
@@ -158,12 +158,14 @@ class Query {
 
   int32_t limit() const;
 
-  const absl::optional<Bound>& start_at() const {
-    return start_at_;
+  const api::optional<Bound> start_at() const {
```

Comment:
You need to change the `start_at_` and `end_at_` member variable to be of type api::optional. Otherwise, the implicit copy constructor will use the wrong absl when compiled in google3.

Reply:
Fun.